custom_components/candy/client/model.py

```diff
@@ -181,28 +185,279 @@ class OvenState(StatusCode):
     HEATING = (1, "Heating")
 
 
+class OvenSelection(StatusCode):
+    IDLE = (0, "Idle")
+    CONVECTION = (3, "Convection")
+    ON_DIFFERENT_LEVELS = (5, "On differend levels")
+    CONVECTION_AND_FAN = (7, "Convection and fan")
+    DEFROSTING = (8, "Defrosting")
+    HEATING = (9, "Heating")
+    GRILL = (11, "Grill")
+    PIZZA = (13, "Pizza")
+    BOTTOM_HEATER = (14, "Bottom heater")
+    GRILL_AND_FAN = (19, "Grill and fan")
+
+
 @dataclass
 class OvenStatus:
     machine_state: OvenState
     program: int
-    selection: int
+    selection: OvenSelection
     temp: float
+    tempSet: float
     temp_reached: bool
     program_length_minutes: Optional[int]
     remote_control: bool
+    remaining_minutes: int
 
     @classmethod
     def from_json(cls, json):
         return cls(
             machine_state=OvenState.from_code(int(json["StartStop"])),
-            program=int(json["Program"]),
-            selection=int(json["Selettore"]),
-            temp=round(fahrenheit_to_celsius(int(json["TempRead"]))),
+            program=int(only_numerics(json["Program"])),
+            selection=OvenSelection.from_code(int(only_numerics(json["Selettore"]))),
+            temp=round(fahrenheit_to_celsius(int(json["TempRead"])))
+            if not json["Program"].startswith("P")
+            else int(json["TempRead"]) / 10,
+            tempSet=round(fahrenheit_to_celsius(int(json["TempSet"])))
```

Comment:
This should rather be `temp_set` according to Python standards

Reply:
Ok, thanks I changed in source.